### PR TITLE
Edit form

### PR DESCRIPTION
**Pull request by AurelieFqt**

### CODE REVIEW

**Review by KevinRoig — APPROVED** [review]
ok

File: src/Controller/GalleryController.php [hunk 1]
```diff
@@ -31,7 +33,52 @@ public function showImagesByCategorie(string $categorie, ImageRepository $imageR
     }
 
     /**
-     * @Route("/gallerie-passages-secrets", name="gallerie-passages-secrets")
+     * @Route("/gallerie-design-metallique/{categorie}/{id}",
```
Comment (RomainClair) [inline src/Controller/GalleryController.php:36]: The categorie parameter seems unnecessary

File: src/Controller/GalleryController.php [hunk 2]
```diff
@@ -31,7 +33,52 @@ public function showImagesByCategorie(string $categorie, ImageRepository $imageR
...
+    /**
+     * @Route("/gallerie-design-metallique/{categorie}/{id}/edit",
+     *      name="gallery_metallicDesign_category_showImg_edit", methods={"GET","POST"})
+     */
+    public function edit(Request $request, Image $image): Response
+    {
+        $form = $this->createForm(EditFormType::class, $image);
+        $form->handleRequest($request);
+        if ($form->isSubmitted() && $form->isValid()) {
+            $this->getDoctrine()->getManager()->flush();
```
Comment (RomainClair) [inline src/Controller/GalleryController.php:55]: It is a symfony best practice to use dependency injection for services (like the EntityManager)

File: src/Controller/GalleryController.php [hunk 3]
```diff
@@ -31,7 +33,52 @@ public function showImagesByCategorie(string $categorie, ImageRepository $imageR
...
+
+            return $this->redirectToRoute('gallery_metallicDesign');
+        }
+
+        return $this->render('form/edit.html.twig', [
+            'image' => $image,
```
Comment (RomainClair) [inline src/Controller/GalleryController.php:61]: The image twig parameter seems unnecessary

File: src/Controller/GalleryController.php [hunk 4]
```diff
@@ -31,7 +33,52 @@ public function showImagesByCategorie(string $categorie, ImageRepository $imageR
...
+    /**
+     * @Route("/gallerie-design-metallique/{categorie}/{id}/delete",
+     *          name="gallery_metallicDesign_category_showImg_delete", methods={"DELETE"})
+     */
+    public function delete(Request $request, Image $image): Response
+    {
+        if ($this->isCsrfTokenValid('delete' . $image->getId(), $request->request->get('_token'))) {
+            $entityManager = $this->getDoctrine()->getManager();
+            $entityManager->remove($image);
+            $entityManager->flush();
```
Comment (RomainClair) [inline src/Controller/GalleryController.php:75]: Once again : It is a symfony best practice to use dependency injection for services (like the EntityManager)

**Review by RomainClair — CHANGES_REQUESTED** [review]
You probably shouldn't delete migrations already in the master or dev branch
Your Image entity should probably have constraints for the symfony validator to do it's job.
For example, you should check that the category is on of the possible choices

**Pull request closed without merging**